File: Chess/src/main/java/game/piece/Coordinate.java
package game.piece;

public class Coordinate implements java.io.Serializable {
    public final int x;
    public final int y;

    public Coordinate(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    @Override
    public String toString() {
        return "[X:" + x + ", Y:" + y + "]";
    }

    @Override
    public boolean equals(Object obj) {
        return (((Coordinate) obj).getX() == this.x && ((Coordinate) obj).getY() == this.y);
    }

    public Coordinate plus(Coordinate coord) {
        return new Coordinate(this.x + coord.getX(), this.y + coord.getY());
    }
}

File: Chess/src/main/java/game/piece/Coordinate.kt
package game.piece

import java.io.Serializable

class Coordinate(@JvmField val x: Int, @JvmField val y: Int) : Serializable {

    override fun toString(): String {
        return "[X:$x, Y:$y]"
    }

    override fun equals(other: Any?): Boolean {
        return ((other as Coordinate?)!!.x == x) && (other!!.y == y)
    }

    operator fun plus(coord: Coordinate): Coordinate {
        return Coordinate(x + coord.x, y + coord.y)
    }
}